File: leetcode/0190-reverse-bits/Solution.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int reversed = 0;
        int leftShift = 31;
        while (n != 0) {
            if ((n & 1) == 1) {
                reversed += 1 << leftShift;
            }
            leftShift--;
            n = n >>> 1;
        }
        
        return reversed;
    }
}

File: leetcode/0190-reverse-bits/solution.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int rev = 0;
        int i = 0;
        while (i < 32) {
            if (n & 1) {
                rev += 1 << (31 - i);
            }
            n = n >> 1;
            i++;
        }
        return rev;
    }
};